File: Assignment 5/main.py
class Time:

    def __init__(self):
        self.hours = 12
        self.minutes = 0
        self.AM_PM = 'A'

    def getHours(self):
        return self.hours

    def getMinutes(self):
        return self.minutes

    def getAM_PM(self):
        return self.AM_PM

    def setTime(self, hours, minutes, AM_PM):
        if (hours >= 1 and hours <= 12):
            self.hours = hours
        else:
            self.hours = 12
        if (minutes >= 0 and minutes <= 59):
            self.minutes = minutes
        else:
            self.minutes = 0
        if (AM_PM == 'A' or AM_PM == 'P'):
            self.AM_PM = AM_PM
        else:
            self.AM_PM = 'A'


Test1 = Time()

print("testing default")
print(Test1.getHours())
print(Test1.getMinutes())
print(Test1.getAM_PM())

print("testing setTime")
Test1.setTime(2, 40, 'P')
print(Test1.getHours())
print(Test1.getMinutes())
print(Test1.getAM_PM())

print("testing invalid inputs")
Test1.setTime(25, 102, 'Z')
print(Test1.getHours())
print(Test1.getMinutes())
print(Test1.getAM_PM())

print("testing user input")
Test1.setTime(int(input("H\n")), int(input("M\n")), input("AP\n"))
print(Test1.getHours())
print(Test1.getMinutes())
print(Test1.getAM_PM())

File: Assignment 5/Main.cpp
#include <new>
#include <string>
#include <iostream>

using namespace std;


class Time
{
public:
	// Constructors
	Time();
	Time(int hours, int minutes, char AM_PM);
	Time(const Time &org); // copy constructor
	// Destructor
	~Time();

	// accessors
	int getHours() const;
	int getMinutes() const;
	char getAMPM() const;

	// mutators
	void setTime(int hours, int minutes, char AM_PM);
	// overloaded operators member functions go here


private:
	// member variables
	int *hours;
	int minutes;
	char AM_PM;
};

int main() {

	Time testTime;
	Time copyTime;

	cout << "testing default" << endl;
	cout << testTime.getHours() << endl;
	cout << testTime.getMinutes() << endl;
	cout << testTime.getAMPM() << endl;
	cout << endl;

	cout << "testing setTime" << endl;
	testTime.setTime(3, 55, 'P');
	cout << testTime.getHours() << endl;
	cout << testTime.getMinutes() << endl;
	cout << testTime.getAMPM() << endl;
	cout << endl;

	cout << "testing copy" << endl;
	copyTime = testTime;
	cout << copyTime.getHours() << endl;
	cout << copyTime.getMinutes() << endl;
	cout << copyTime.getAMPM() << endl;
	cout << endl;

	cout << "testing invalid inputs" << endl;
	testTime.setTime(45, 129, 'Z');
	cout << testTime.getHours() << endl;
	cout << testTime.getMinutes() << endl;
	cout << testTime.getAMPM() << endl;
	cout << endl;

	cout << "testing user inputs" << endl;
	int x = 0;
	int y = 0;
	char z = 0;
	cin >> x;
	cin >> y;
	cin >> z;
	testTime.setTime(x, y, z);
	cout << testTime.getHours() << endl;
	cout << testTime.getMinutes() << endl;
	cout << testTime.getAMPM() << endl;
	cout << endl;

	return 0;
}

Time::Time() : minutes(0), AM_PM('A')
{
	hours = new int;
	*hours = 12;
}

Time::Time(int h, int m, char AP)
{
	hours = new int;
	setTime(h, m, AP);
}

Time::Time(const Time &org) : minutes(org.minutes), AM_PM(org.AM_PM)
{
	hours = new int;
	*hours = *org.hours;
}

Time::~Time()
{
	delete hours;
}

int Time::getHours() const
{
	return *hours;
}

int Time::getMinutes() const
{
	return minutes;
}

char Time::getAMPM() const
{
	return AM_PM;
}

void Time::setTime(int h, int m, char AP)
{
	if (h >= 1 && h <= 12)
	{
		*hours = h;
	}
	else
	{
		*hours = 12;
	}
	if (m >= 0 && m <= 59)
	{
		minutes = m;
	}
	else
	{
		minutes = 0;
	}
	if (AP == 'A' || AP == 'P')
	{
		AM_PM = AP;
	}
	else
	{
		AM_PM = 'A';
	}
}